pkg/prompts/prompts.go

```diff
@@ -244,3 +272,40 @@ func Select[T any](label string, items []T, opt *SelectOpt[T]) (T, error) {
 
 	return items[i], nil
 }
+
+func getCurrentDirName() (string, error) {
+	dir, err := os.Getwd()
+	if err != nil {
+		log.Fatalf("getting current directory: %v", err)
+	}
+	return filepath.Base(dir), nil
+}
+
+// Sanitize the directory name to comply with k8s label rules
+func sanitizeAppName(name string) string {
+	// Remove all characters except alphanumeric, '-', '_', '.'
+	reg := regexp.MustCompile(`[^a-zA-Z0-9-_.]+`)
+	sanitized := reg.ReplaceAllString(name, "")
+
+	// Trim leading and trailing '-', '_', '.'
+	sanitized = strings.Trim(sanitized, "-._")
+
+	// If name is empty, return empty string
+	if sanitized == "" {
```

Comment:
what does this early check do?

Reply:
if the user input is empty it returns an entry string and in the below code I'm setting the default value which will be the directory name
if customPrompt.Name == "APPNAME" {
		// Sanitize the user input
		sanitizedInput := sanitizeAppName(input)
		if sanitizedInput == "" {
			// If sanitized input is empty, use the sanitized directory-based name
			sanitizedInput = defaultValue
		}
		input = sanitizedInput
	}

Reply:
but why in this particular place? You could have this same check everywhere you reassign sanitized

if the entire function runs "" will still be returned.

Reply:
I thought it would be good to just return empty string without executing the remaining logic, but yeah it may be redundant. I will update it.